src/hooks/useBasisForm/validation/validateInput.ts

```diff
@@ -0,0 +1,23 @@
+export const NUMERIC_REGEX = /^\d*$/;
+export const DECIMAL_REGEX = /^\d*(\.\d{2})?$/;
+
+export const validateInput = (
+  value: string,
+  inputProps: Record<string, any>
+) => {
+  const { optional, variant } = inputProps;
+
+  if (!value && !optional) {
```

Comment:
Looking at our [current handling for optional fields](https://github.com/moroshko/basis/blob/ba9469e6bacce85275f0b5ee10ec6efb365d80fa/src/components/Form.js#L164-L167):
* Does `react-hook-form` know not to run the validation function when the input is disabled, or should we handle it in basis?
* When the field is optional, we want to proceed with the extra checks below **only if the field is not empty/blank.**
* Didn't we agree to leave the `trim()` functionality?